Mathlib/NumberTheory/LegendreSymbol/JacobiSymbol.lean

```diff
@@ -501,3 +503,160 @@ theorem mod_right (a : ℤ) {b : ℕ} (hb : Odd b) : J(a | b) = J(a | b % (4 * a
 end jacobiSym
 
 end Jacobi
+
+
+section FastJacobi
+
+/-!
+### Fast computation of the Jacobi symbol
+We follow the implementation as in `Mathlib.Tactic.NormNum.LegendreSymbol`.
```

Comment:
Please describe the algorithm in a few sentences here or in the module docstring. Also, is it possible to deduplicate the code between this file and the `NormNum` extension?

Reply:
I don't know enough about metaprogramming in Lean to deduplicate the code (when I did a naive replacement I ran into a deep recursion error). I suspect the usual strategy is that there is a split between the computation rule and norm_num extension which additionally focuses on producing small proof terms. For example there is also a lot of duplicate logic between Nat.fastFib and NormNum/NatFib.